### ISSUE TITLE
Rich deposit triggers another rich deposit by chance

### ISSUE THREAD

**Issue by Minequirion** [Th3Shadowbroker/OuroborosMines#11]
**Describe the bug**
When a rich deposit is replaced by the own blocktype and then mined again, it seems that the new ore-block can trigger another rich deposit.
Therefore the block can be mined more often than set in the configuration.
In one specific test a coal-block could be mined up to 18 times even if the configuration was set to a maximum of 5 additional times.

**Testing environment**
The following region configuration was used for this test:
```yml
inherit: false
materials:
  stone:
    replacements:
    - cobblestone
    cooldown: 10
    
  coal_ore:
    replacements:
    - stone
    cooldown: 20-170
    rich-chance: 75
    rich-amount: 1-5
```

**To Reproduce**
1. Setup a region with the configuration above
2. Mine coal deposits until a rich deposit is found
3. Count the amount of mineable blocks after the first one

**Expected behavior**
The additional mineable amount of a rich deposit should not exceed the maximum value set under "rich-amount" in the configuration file.
A spawned rich deposit should not trigger another rich deposit.

**Desktop (please complete the following information):**
 - Minecraft-version: 1.14.4
 - Server-version: git-Spigot-56f8471-7554e08 (Implementing API version 1.14.4-R0.1-SNAPSHOT)

**Comment by Th3Shadowbroker** [comment]
Thanks for the report, I'll look into this 👍

**Comment by Th3Shadowbroker** [comment]
So I were able to reproduce the problem and fixed the bug in #14.
The problem was, that the "richness" was redrawn at 0 which could lead to chained rich deposits.
